### PR TITLE
PYTHON-258  tests error and negative scenarios

### PR DESCRIPTION
**Pull request by GregBestland**
Modified existing tests to make sure 0 results work correctly, and exceptions are thrown on non-indexed collections.
PYTHON-258

### CODE REVIEW

File: tests/integration/cqlengine/query/test_queryset.py [hunk 1]
```diff
@@ -266,6 +271,10 @@ def setUpClass(cls):
                                         test_result=45, test_list=[1, 2, 3], test_set=set([1, 2, 3]),
                                         test_map={'1': 1, '2': 2, '3': 42})
 
+        IndexedTestModel.objects.create(test_id=14, attempt_id=4, description='list14', expected_result=75,
```
Comment (aholmberg) [inline tests/integration/cqlengine/query/test_queryset.py]: It's not really making a difference the way the tests are written, but I noticed ``test_id`` is not incremented like the previous entries.

**Comment by aholmberg** [comment]
+1

File: tests/integration/cqlengine/query/test_queryset.py (hunk 1 above)
Comment (aboudreault) [inline tests/integration/cqlengine/query/test_queryset.py]: We will mostly forgot that many times. I would that whenever we need an id, we use a test class function. ie. test_next_id()

**Comment by GregBestland** [comment]
Fixed Id issue.

**Pull request merged**